docs.feldera.com/docs/pipelines/modifying.md

```diff
@@ -0,0 +1,436 @@
+
+### Limitation 3: Table evolution is not yet supported
+
+Adding, removing or renaming a table column will clear the table's state and require the table
+to be ingested from scratch.
+
+## API
+
+The Feldera REST API contains several elements that support the bootstrapping feature described in this document:
+
+* The `bootstrap_policy` argument to the [`/start` endpoint](https://docs.feldera.com/api/stop-the-pipeline-asynchronously-by-updating-the-desired-state). This argument specifies how the pipeline should
+  behave when the checkpointed version of the program differs from the current version. The supported values are
+  `await_approval`, `allow`, and `reject`. The default value is `await_approval`. See the
+  [Restart the pipeline](#3-restart-the-pipeline) section above for details.
+
+* Two new runtime states reported by the [pipeline status endpoint](https://docs.feldera.com/api/retrieve-a-pipeline) in the `deployment_runtime_status` field:
+  * `AwaitingApproval` - When starting the pipeline with `bootstrap_policy=await_approval`, if the pipeline
+    requires bootstrapping, it will stop in the `AwaitingApproval` state waiting for the user to approve the
+    changes. While in this state, the `deployment_runtime_status_details` field lists added, modified, and removed
+    tables, views, and connectors in JSON format.
+    The pipeline exits this state when the user either approves the pipeline via the `/approve` API
+    or force-stops the pipeline with `/stop?force=true`.
+  * `Bootstrapping` - In this state the pipeline evaluates new and modified views. Once bootstrapping completes,
+    the pipeline automatically moves into the `Running` or `Paused` state.
+
+* The `/approve` endpoint moves the pipeline from the `AwaitingApproval` to `Bootstrapping` state.
+
+## How different types of changes are handled
+
+This section describes how the bootstrapping process handles different types of changes
+to your pipeline.
+
+### Views
+
+#### Adding a view
+
+New views are evaluated during bootstrapping; their full contents are sent to all attached output connectors.
+
+#### Modifying a view
+
+A view is modified when the user changes its SQL definition or as a side effect of a change to an upstream
+view or a runtime upgrade. The old state of modified views is discarded; the view is evaluated from
+scratch, and its full contents are sent to all attached output connectors.
+
+#### Removing a view
+
+The view, all of its connectors, and all internal indexes used to maintain the view incrementally
+are discarded.
+
+### Tables
+
+#### Adding a table
+
+New tables are created empty.
+Note that renaming an existing table is treated as deleting the old table and creating a new one.
+
+#### Modifying a table
+
+Modifying a table clears the state of the table and all views that depend on it during bootstrapping.
+All table connectors are reset to their initial states.
+
+#### Removing a table
+
+All state of the deleted table is discarded.
```

Comment:
clearly, compiling a program that has views depending on this table will also fail